### PR TITLE
feat(ui): Add "Create Team Modal"

### PR DESCRIPTION
**Pull request by billyvg**
* Change "project teams" view to use create team modal flow and to add team
to project after creation
* Change Organization Teams View to use create team modal flow
* Fix deprecation warnings with getEndpoint
* Refactor `<AllTeamsList>` to be able to use create team modal if necessary
* feat(ui): Add "create team modal" + action creator
* Refactor form from `teamCreate`, change browser direct to use react-router
* ref(ui): "NarryLayout" --> "NarrowLayout"
* Add fingerprint to webpack chunk load error
* Add `bypass` prop to `<Confirm>` modal to skip modal
* Add `createTeam` action creator
* Add `addTeamToProject` action creator
* Wrap form children so that last-child works when there is a footer
* Add padding prop to fieldWrapper
* feat(ui): Add "hideControlState" prop to `<FormField>`

Adds a create team modal on Organization --> Teams:
image

Adds "create team" button on Projects --> Teams (as well as using new create team modal flow). Creating a team here will also add that team to the current project in scope:
image

### CODE REVIEW

**Comment by dcramer** [comment]
any way to make the text input full width?

**Comment by billyvg** [comment]
Ya input is now fill width unless an error occurs.

**Comment by MaxBittker** [comment]
image

**Comment by MaxBittker** [comment]
when used from the teams list, I think it should maybe redirect to the team settings?

File: src/sentry/static/sentry/app/actionCreators/teams.jsx [hunk 1]
```diff
@@ -96,6 +96,38 @@ export function leaveTeam(api, params, options) {
...
+            organization: params.orgId,
+          })
+        );
+        return data;
+      },
+      err => {
+        TeamActions.createTeamError(team.slug || team.name, err);
+        addErrorMessage(
+          tct('Unable to create [team] in the [organization] organization', {
+            team: `#${team.slug || team.name}`,
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/actionCreators/teams.jsx]: when would just name get passed?

File: src/sentry/static/sentry/app/components/createTeam/createTeamForm.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  render() {
+    let {organization} = this.props;
+    let features = new Set(organization.features);
+
+    return (
+      <React.Fragment>
+        <p>
+          {t(
+            "Teams group members' access to a specific focus, e.g. a major product or application that may have sub-projects."
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/createTeam/createTeamForm.jsx]: is this copy pasted? it seems a bit outdated. 
maybe you could say 
"Teams group members' access to specific projects, and can be assigned to issues"

File: src/sentry/static/sentry/app/components/lazyLoad.jsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ class LazyLoad extends React.Component {
   handleFetchError = error => {
     // eslint-disable-next-line no-console
     console.error(error);
-    Raven.captureException(error);
+    Raven.captureException(error, {fingerprint: ['webpack', 'error loading chunk']});
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/lazyLoad.jsx]: 👍

File: src/sentry/static/sentry/app/components/modals/createTeamModal.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    closeModal: PropTypes.func,
+    onClose: PropTypes.func,
+    Body: PropTypes.oneOfType([PropTypes.func, PropTypes.node]).isRequired,
+    Header: PropTypes.oneOfType([PropTypes.func, PropTypes.node]).isRequired,
+    organization: SentryTypes.Organization.isRequired,
+    project: SentryTypes.Project,
+  };
+
+  handleSubmit = data => {
+    createTeam(new Client(), data, {orgId: this.props.organization.slug}).then(
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/modals/createTeamModal.jsx]: should these action creators just manage their own api client now?

File: src/sentry/static/sentry/app/stores/teamStore.jsx [hunk 5]
```diff
@@ -51,6 +52,10 @@ const TeamStore = Reflux.createStore({
     this.loadInitialData(this.items.filter(team => team.slug !== slug));
   },
 
+  onCreateSuccess(team) {
+    this.loadInitialData([...this.items, team]);
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/stores/teamStore.jsx]: why does everything in here use loadInitialData? the extra triggering seems bad

File: src/sentry/static/sentry/app/views/settings/components/forms/field/fieldControl.jsx [hunk 6]
```diff
@@ -45,13 +45,13 @@ class FieldControl extends React.Component {
     alignRight: PropTypes.bool,
     disabled: PropTypes.bool,
     disabledReason: PropTypes.node,
-    hideControlState: PropTypes.bool,
+    flexibleControlStateSize: PropTypes.bool,
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/settings/components/forms/field/fieldControl.jsx]: this name was confusing to me - it means that the size is set by the parent?

File: tests/js/spec/views/projectTeams.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+    Client.clearMockResponses();
+  });
+
+  it('renders', function() {
+    let wrapper = shallow(
+      <ProjectTeams
+        params={{orgId: org.slug, projectId: project.slug}}
+        organization={org}
+      />,
+      {
```
Comment (MaxBittker) [inline tests/js/spec/views/projectTeams.spec.jsx]: can you use the `TestStubs.routerContext` here?

File: tests/js/spec/views/projectTeams.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+      })
+    );
+
+    // Remove second team
+    wrapper
+      .find('PanelBody Button')
+      .first()
+      .simulate('click');
+
+    // Modal opens because this is the last team in project
```
Comment (MaxBittker) [inline tests/js/spec/views/projectTeams.spec.jsx]: nice test 👍

File: tests/js/spec/views/teamCreate.spec.jsx [hunk 9]
```diff
@@ -20,13 +20,17 @@ describe('TeamCreate', function() {
     });
   });
 
-  describe('onSubmitSuccess()', function() {
+  describe('handleSubmitSuccess()', function() {
     let wrapper;
-    let locationAssignMock;
+    let redirectMock = jest.fn();
```
Comment (MaxBittker) [inline tests/js/spec/views/teamCreate.spec.jsx]: redirect is a common thing to need to mock out - maybe we could make this a bit easier to do using the router fixture? what you did looks good though

**Review by MaxBittker — APPROVED** [review]
I'm slightly worried about the UX of not prompting people to associate users in these modals (the way it links the project is really nice), but it works smoothly and code looks good.

maybe we could mention the team settings page in the copy?

File: src/sentry/static/sentry/app/actionCreators/teams.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/actionCreators/teams.jsx]: slug/name is behind a feature tag

File: src/sentry/static/sentry/app/components/modals/createTeamModal.jsx (hunk 4 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/modals/createTeamModal.jsx]: I'm not sure right now, I wanted to keep the API the same between the team action creators (they all accept api client as the first arg). If they were to handle their own API client, components lose the ability to easily cancel API requests (how often this happens in practice should probably be measured).

File: src/sentry/static/sentry/app/views/settings/components/forms/field/fieldControl.jsx (hunk 6 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/forms/field/fieldControl.jsx]: mmm yeah I wanted the opposite of `fixed` size. It means that control state doesn't have a fixed width.

File: tests/js/spec/views/teamCreate.spec.jsx (hunk 9 above)
Comment (billyvg) [reply tests/js/spec/views/teamCreate.spec.jsx]: Hmm that's true, currently the router fixture uses `sinon.spy`, we could update to use jest mock functions so we can use jest assertions.

File: src/sentry/static/sentry/app/stores/teamStore.jsx (hunk 5 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/stores/teamStore.jsx]: It's not an "extra" trigger though? It is only called once.

**Comment by billyvg** [comment]
Re: Copy + Members, let's handle that in follow-ups after people get a chance to use it first

**Pull request merged**